activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -641,8 +658,8 @@ def load_types_queries(initializer, oids)
             yield query + "WHERE t.oid IN (%s)" % oids.join(", ")
           else
             yield query + initializer.query_conditions_for_known_type_names
-            yield query + initializer.query_conditions_for_known_type_types
             yield query + initializer.query_conditions_for_array_types
+            yield query + initializer.query_conditions_for_known_type_types
```

Comment:
The result of running
```ruby
initializer.run(known_type_names_records)
initializer.run(known_type_types_records)
initializer.run(array_types_records)
```
is different from runing
```ruby
initializer.run(known_type_names_records + known_type_types_records + array_types_records)
```
The second one is the orignal behaviour before https://github.com/rails/rails/pull/40876. Adjust the order of the queries would fix the inconsistent issue.

Reply:
Also the adjusted order matches https://github.com/rails/rails/blob/beed933b9bb9f317697a95a36da067a67797e0d8/activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb#L28-L32